### PR TITLE
Uses the global store for the post title and adds undo/redo support.

### PR DESCRIPTION
**Pull request by diegoreymendez**
## Description:

Following @Tug 's indications [here](
https://github.com/wordpress-mobile/gutenberg-mobile/pull/450#discussion_r248672329), this PR wires our title to the global store title.

This PR also adds undo/redo support for the title.

Moves forward #372 .

## Dependencies:

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/13514

### Testing edits:

1. Launch the demo app.
2. Make sure the initial title is properly set up.
3. Edit the title, switch to HTML mode and back.

Make sure the title is still as you left it before switching modes.

### Testing undo redo:

1. Edit the title as you see fit.
2. Use the undo / redo buttons from the toolbar.

Make sure undo / redo works as it does in web-Gutenberg (ie: that's our baseline for comparing).

### CODE REVIEW

**Comment by diegoreymendez** [comment]
Fyi, @daniloercoli

**Comment by diegoreymendez** [comment]
Reassigned to @daniloercoli since he's working on the title feature with me.

File: src/app/AppContainer.js [hunk 1]
```diff
@@ -92,7 +91,7 @@ class AppContainer extends React.Component<PropsType, StateType> {
 				toggleHtmlModeAction={ this.toggleHtmlModeAction }
 				setTitleAction={ this.setTitleAction }
```
Comment (pinarol) [inline src/app/AppContainer.js]: Hey @diegoreymendez 👋 it looks to me that we can also get rid of passing this.setTitleAction as a prop from component to component. So other components can also use editPost( { title: title } ) to via their own props with the help of withDispatch. what do you think?

Comment (diegoreymendez) [reply src/app/AppContainer.js]: That's a good observation.  Thanks @pinarol!

Considering this PR is working as expected right now though, I've created [an issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/512) that I will be tackling right after this is complete.

As usual my goal is to split the work into atomic steps where necessary.

**Pull request merged**